rspirv/sr/module.rs

```diff
@@ -18,8 +18,18 @@ pub struct EntryPoint {
 
 #[derive(Debug)]
 pub struct BasicBlock {
-   pub ops: Vec<Op>,
-   pub terminator: ops::Terminator,
+    pub arguments: Vec<Token<Type>>,
```

Comment:
I really like this approach to basic blocks, it's lovely and get's rid of a load of oddities surrounding phi instructions.

Reply:
Although thinking about this more isn't this going to make instructions that use these much more complicated as they can't just refer to a previous operation? We may want to introduce an SSA Value type, it would also get ridd of all the weirdness of having to store instructions in Storages.

Reply:
I'm not sure what you mean by "refer to a previous operation", please clarify?

As for SSA Value, please file an issue and describe what you have in mind.

Reply:
Currently, operations use `Token<Op>` to refer to the results of previous instructions? With this change, it's impossible for an instruction to refer to the result of a phi instruction as it isn't stored in the `Storage<Op>` any more. The simplest solution would be to just add some dummy ops to the storage.

Reply:
Correct. We'll still have some `Op::Phi` things in the storage.

Reply:
+1 to what @Skepfyr said. I've also mentioned that we don't have a proper SSA value data type in structured representation at various places. (Also because the structured representation is under development right now so it's broken anyway. ;-P) Depending on how we want to model function parameters and Phi nodes (they are both instructions in SPIR-V spec and DR), we may end up having two more cases an instruction can reference as an operand. So having a value type, which is an enum among FunctionArgument, BlockArgument, and OpResult. And I can see this is how the current impl is heading for. (We baked function argument into `sr::Function` and block argument to `sr::Block`.) Another thing we need to think about is how to support def-use chain. Normally the use chain is associated with the value.